File: Java/Program/Komputer.java
import java.util.ArrayList;
import java.util.List;

public class Komputer {
    private List<CPU> cpus = new ArrayList<>();
    private List<RAM> rams = new ArrayList<>();
    private List<Harddrive> harddrives = new ArrayList<>();
    private List<GPU> gpus = new ArrayList<>();
    private PowerSupply psu;
    private String os;

    public Komputer(String os) {
        this.os = os;
    }

    public void setPSU(PowerSupply psu) {
        this.psu = psu;
    }

    public PowerSupply getPSU() {
        return psu;
    }

    public void addCPU(CPU cpu) {
        cpus.add(cpu);
    }

    public void addRAM(RAM ram) {
        rams.add(ram);
    }

    public void addHardDrive(Harddrive harddrive) {
        harddrives.add(harddrive);
    }

    public void addGPU(GPU gpu) {
        gpus.add(gpu);
    }

    public void bootUp() {
        System.out.println("\n[ Booting " + os + " ]");
        if (psu == null) {
            System.out.println("No PSU installed!");
            return;
        }
        psu.turnOn();
        System.out.println("System Ready!");
    }

    public void shutDown() {
        System.out.println("\n[ Shutting Down " + os + " ]");
        if (psu != null) {
            psu.turnOff();
        }
        System.out.println("System Off!");
    }

    public void getSystemInfo() {
        System.out.println("\n==== CONFIGURATION ====");
        System.out.println("OS: " + os);
        for (CPU cpu : cpus) {
            System.out.print("CPU: ");
            cpu.getInfo();
        }
        for (GPU gpu : gpus) {
            System.out.print("GPU: ");
            gpu.getInfo();
        }
        for (RAM ram : rams) {
            System.out.print("RAM: ");
            ram.getInfo();
        }
        for (Harddrive harddrive : harddrives) {
            System.out.print("HDD: ");
            harddrive.getInfo();
        }
        System.out.println("PSU: " + (psu != null ? "Installed" : "Not Installed"));
        System.out.println("=======================");
    }
}

File: Python/Program/Komputer.py
from CPU import CPU
from RAM import RAM
from Harddrive import Harddrive
from PowerSupply import PowerSupply
from GPU import GPU

class Komputer:
    def __init__(self, os=""):
        self.cpus = []
        self.rams = []
        self.harddrives = []
        self.gpus = []
        self.psu = None
        self.os = os
    
    def set_psu(self, p):
        self.psu = p
    
    def get_psu(self):
        return self.psu
    
    def add_cpu(self, c):
        self.cpus.append(c)
    
    def add_ram(self, r):
        self.rams.append(r)
    
    def add_hard_drive(self, h):
        self.harddrives.append(h)
    
    def add_gpu(self, g):
        self.gpus.append(g)
    
    def boot_up(self):
        print(f"\n[ Booting {self.os} ]")
        if not self.psu:
            print("No PSU installed!")
            return
        self.psu.turn_on()
        print("System Ready!")
    
    def shut_down(self):
        print(f"\n[ Shutting Down {self.os} ]")
        if self.psu:
            self.psu.turn_off()
        print("System Off!")
    
    def get_system_info(self):
        print("\n==== CONFIGURATION ====")
        print(f"OS: {self.os}")
        for i in range(len(self.cpus)):
            print("CPU: ", end="")
            self.cpus[i].get_info()
        for i in range(len(self.gpus)):
            print("GPU: ", end="")
            self.gpus[i].get_info()
        for i in range(len(self.rams)):
            print("RAM: ", end="")
            self.rams[i].get_info()
        for i in range(len(self.harddrives)):
            print("HDD: ", end="")
            self.harddrives[i].get_info()
        print(f"PSU: {'Installed' if self.psu else 'Not Installed'}")
        print("=======================")
